leshan-client-cf/src/main/java/org/eclipse/leshan/client/californium/impl/CaliforniumEndpointsManager.java

```diff
@@ -97,36 +97,9 @@ public synchronized Server createEndpoint(ServerInfo serverInfo) {
         if (serverInfo.isSecure()) {
             Builder newBuilder = new Builder(dtlsConfigbuilder.getIncompleteConfig());
 
-            // Support PSK
-            if (serverInfo.secureMode == SecurityMode.PSK) {
-                StaticPskStore staticPskStore = new StaticPskStore(serverInfo.pskId, serverInfo.pskKey);
-                newBuilder.setPskStore(staticPskStore);
-                serverIdentity = Identity.psk(serverInfo.getAddress(), serverInfo.pskId);
-            } else if (serverInfo.secureMode == SecurityMode.RPK) {
-                // set identity
-                newBuilder.setIdentity(serverInfo.privateKey, serverInfo.publicKey);
-                // set RPK truststore
-                final PublicKey expectedKey = serverInfo.serverPublicKey;
-                newBuilder.setRpkTrustStore(new TrustedRpkStore() {
-                    @Override
-                    public boolean isTrusted(RawPublicKeyIdentity id) {
-                        PublicKey receivedKey = id.getKey();
-                        if (receivedKey == null) {
-                            LOG.warn("The server public key is null {}", id);
-                            return false;
-                        }
-                        if (!receivedKey.equals(expectedKey)) {
-                            LOG.debug(
-                                    "Server public key received does match with the expected one.\nReceived: {}\nExpected: {}",
-                                    receivedKey, expectedKey);
-                            return false;
-                        }
-                        return true;
-                    }
-                });
-                serverIdentity = Identity.rpk(serverInfo.getAddress(), expectedKey);
-            } else if (serverInfo.secureMode == SecurityMode.OSCORE) {
-                System.out.println("Adding OSCORE CTX " + serverInfo.getFullUri().toASCIIString());
+            // oscore only mode
+            if(serverInfo.useOscore) {
+            	System.out.println("Adding OSCORE CTX " + serverInfo.getFullUri().toASCIIString());
```

Comment:
Please use LOG instead of sysout ? :)

Reply:
Yes, that was an old debug statement I should have fixed. Now done in commit a83918e1619c160747895ccc6e44731d1d90a6ed